src/main.cpp

```diff
@@ -180,857 +219,1155 @@ void forwardIt(const char *line);
+    json["MCUtype"] = systemSettings.MCUtype;
+    json["breakoutGeneration"] = systemSettings.breakoutGeneration;
+    json["bootloader"] = systemSettings.bootloader;
+
+    serializeJson(json, *response);
+#ifdef CORS_HEADER
+    response->addHeader("Access-Control-Allow-Origin", "*");
+#endif
+    request->send(response);
+    Serial2.print("sys?\r\n");
+}
 
-  serializeJson(json, *response);
-  request->send(response);
+void notFound(AsyncWebServerRequest *request) {
+    request->send(404, "text/plain", "Not found");
 }
 
-void handleStation(AsyncWebServerRequest *request)
-{
-
-  if (request->hasParam(PARAM_MMSI) && request->hasParam(PARAM_CALLSIGN) && request->hasParam(PARAM_VESSELNAME) && request->hasParam(PARAM_VESSELTYPE) && request->hasParam(PARAM_LOA) && request->hasParam(PARAM_BEAM) && request->hasParam(PARAM_PORTOFFSET) && request->hasParam(PARAM_BOWOFFSET))
-  {
-    stationSettings.mmsi = request->getParam(PARAM_MMSI)->value();
-    stationSettings.callsign = request->getParam(PARAM_CALLSIGN)->value();
-    stationSettings.vesselname = request->getParam(PARAM_VESSELNAME)->value();
-    stationSettings.vesseltype = request->getParam(PARAM_VESSELTYPE)->value().toInt();
-    stationSettings.loa = request->getParam(PARAM_LOA)->value().toInt();
-    stationSettings.beam = request->getParam(PARAM_BEAM)->value().toInt();
-    stationSettings.bowoffset = request->getParam(PARAM_BOWOFFSET)->value().toInt();
-    stationSettings.portoffset = request->getParam(PARAM_PORTOFFSET)->value().toInt();
-
-    Serial2.println("station " + stationSettings.mmsi + "," + stationSettings.mmsi + "," + stationSettings.callsign + "," + stationSettings.vesselname + "," + stationSettings.vesseltype + "," + stationSettings.loa + "," + stationSettings.beam + "," + stationSettings.bowoffset + "," + stationSettings.portoffset);
-    Serial2.print("station?\r\n");
-  }
-
-  AsyncResponseStream *response = request->beginResponseStream("application/json");
-  DynamicJsonDocument json(1024);
-
-  json["mmsi"] = stationSettings.mmsi;
-  json["callsign"] = stationSettings.callsign;
-  json["vesselname"] = stationSettings.vesselname;
-  json["vesseltype"] = stationSettings.vesseltype;
-  json["loa"] = stationSettings.loa;
-  json["beam"] = stationSettings.beam;
-  json["portoffset"] = stationSettings.portoffset;
-  json["bowoffset"] = stationSettings.bowoffset;
-
-  serializeJson(json, *response);
-  request->send(response);
+void setupWebServer() {
+    server.on("/", HTTP_GET, [](AsyncWebServerRequest *request) {
+        request->send(SPIFFS, "/index.html", "text/html");
+    });
+
+    server.on("/config.html", HTTP_GET, [](AsyncWebServerRequest *request) {
+        request->send(SPIFFS, "/config.html", "text/html");
+    });
+    
+    server.on("/dashboard.html", HTTP_GET, [](AsyncWebServerRequest *request) {
+        request->send(SPIFFS, "/dashboard.html", "text/html");
+    });
+    //    server.serveStatic("/", SPIFFS, "/").setDefaultFile("index.html");
```

Comment:
I was hesitant for that, as I can imagine at some point the config page you may want to password protect. 
If all files are readable the password would be easy redable too.

If all files would be readable directly, than in some ways it may be better to pull the config directly from the json files instead of having specific mout points for them..

Reply:
makes sense, good thoughts